### ISSUE TITLE
Validation should not require VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT_KHR for EXT extension variant

### ISSUE THREAD

**Issue by baldurk** [KhronosGroup/Vulkan-ValidationLayers#1661]
Currently if a buffer is allocated with `VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT` - which is critically is the same as the `VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT_EXT` variant - the validation prints this warning:

> vkBindBufferMemory(): If buffer was created with the VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT_KHR bit set, memory must have been allocated with the VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT_KHR bit set. The Vulkan spec states: If the VkPhysicalDeviceBufferDeviceAddressFeatures::bufferDeviceAddress feature is enabled and buffer was created with the VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT bit set, memory must have been allocated with the VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT bit set (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkBindBufferMemory-bufferDeviceAddress-03339)

But this memory allocate bit only exists in the KHR variant of the extension, and doesn't exist (so isn't required) in the EXT variant. This message should only appear if the KHR extension is in use - or equivalently vulkan 1.2. If the EXT variant is in use on 1.0 or 1.1, the message shouldn't appear.

**Comment by jeffbolznv** [comment]
You're the second person to report this to me today, I'm working on the fix.

FWIW, this was correct when I added the KHR_buffer_device_address validation, but has regressed...

**Comment by dictoon** [comment]
Hi, is this supposed to be fixed in the validation layers that ship with version 1.2.135.0 of the Vulkan SDK?

I'm asking because I'm still getting them with `VK_LAYER_KHRONOS_validation`:

```
Validation Error: [ VUID-vkBindBufferMemory-bufferDeviceAddress-03339 ] Object 0: handle = 0xb09e9c0000000039, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0x44a78781 | vkBindBufferMemory(): If buffer was created with the VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT_KHR bit set, memory must have been allocated with the VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT_KHR bit set. The Vulkan spec states: If the VkPhysicalDeviceBufferDeviceAddressFeatures::bufferDeviceAddress feature is enabled and buffer was created with the VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT bit set, memory must have been allocated with the VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT bit set (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkBindBufferMemory-bufferDeviceAddress-03339)
```

**Comment by Tony-LunarG** [comment]
The commit for #1662 is immediately after the branch for sdk-1.2.135, so no this fix is not in that SDK, but will be in the next one.

**Comment by gg8345** [comment]
The bug is still present in the latest version of the validation layer :  

These lines should have been modified too ?

https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/layers/core_validation.cpp

```
            if (enabled_features.core12.bufferDeviceAddress &&
                (buffer_state->createInfo.usage & VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT) &&
                (!chained_flags_struct || !(chained_flags_struct->flags & VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT))) {
                skip |= LogError(buffer, "VUID-vkBindBufferMemory-bufferDeviceAddress-03339",
                                 "%s: If buffer was created with the VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT bit set, "
                                 "memory must have been allocated with the VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT bit set.",
                                 api_name);
            }
```

**Comment by Tony-LunarG** [comment]
I think that check is correct.  It's saying that if the KHR feature is enabled, the ALLOCATE_DEVICE_ADDRESS_BIT is required, but the lack of enabled_features.buffer_device_address_ext.bufferDeviceAddress in the check is appropriate since the if just the EXT feature is enabled, the requirement doesn't apply.  The layer validation test BufferDeviceAddressEXT tests this case at the very end, where the EXT extension is enabled, memory is allocated without VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT and bound to a buffer that was created with VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT, and no error is expected or found.
